Project management/SEO preparation.md

```diff
@@ -0,0 +1,163 @@
+2. an existing website that we need to migrate and regain or improve their existing search ranking.
+
+### New website
+
+When we are creating a new website, we need to make sure that the **client** is aware of **how the content is structured** on the website. Some of the most important things to be aware of are
+
+- Having different heading sizes in the correct order (H1 -> H2 -> H3 -> H4)
+- No more than one H1 tag
+- Images in the content with set alt tags
+- Enough of inbound and outbound links
+- Set meta descriptions
+- Images shouldn't be too large
+- Set the OG images for the page/post
+
+Don't use SVGs for the featured images or OG images, as they won't get rendered on the social networks and search results
+
+Yoast SEO plugin will warn if these are not properly set, but it's good to have them in mind nonetheless.
+
+Make sure content editors are aware **not to put too large images**. Anything over 200 KB is considered large by Google lighthouse and will be counted as a penalty when calculating a page score.
+
+Besides content, the client should set the proper meta description for every page and post, keywords for the post, and social images (OG images). These can all be set from the Yoast meta box at the bottom of the post/page.
+
+#### Developer notes
+
+Developers should make sure that the correct permalink structure is set (Settings / Permalink Settings - post name), and that all the settings are correctly set up for the Yoast plugin (organization images, social details, Google verification code, etc.).
+
+By default, WordPress will generate a `robots.txt` file that will instruct the crawlers which pages should be crawled. You should add the `sitemap.xml` file to the `robots.txt` list as well. You can do that using the following snippet
+
+```php
+<?php
+
+/**
+ * The file that holds the SeoFunctionality class.
+ *
+ * @package ProjectNamespace\Seo;
+ */
+
+declare(strict_types=1);
+
+namespace ProjectNamespace\Seo;
+
+use ProjectNamespaceVendor\EightshiftLibs\Services\ServiceInterface;
+
+/**
+ * SeoFunctionality class.
+ */
+class SeoFunctionality implements ServiceInterface
+
+{
+
+	/**
+	 * Register all the hooks
+	 *
+	 * @return void
+	 */
+
+	public function register(): void
+
+	{
+		\add_filter('robots_txt', [$this, 'robotsFilter'], 100, 2);
+	}
+
+	/**
+	 * Robots txt modification
+	 *
+	 * Add the sitemap to the robots.
+	 *
+	 * @param string $output The robots.txt output.
+	 * @param bool $public Whether the site is considered "public".
+	 *
+	 * @return string Modified output of the robots.txt file.
+	 */
+
+	public function robotsFilter(string $output, bool $public): string
+	{
+		$homeUrl = get_home_url();
+
+		$output .= "Sitemap: {$homeUrl}/sitemap_index.xml\n";
+
+		return $output;
+	}
+
+}
+
+```
+
+
+This will add the sitemap to the robots file. A sitemap is generated by the Yoast plugin and is a crucial way to tell search engine crawlers which pages should be crawled.

```

Comment:
```suggestion
This will add the sitemap to the robots file. The sitemap is generated by the Yoast plugin and is a crucial way to tell search engine crawlers which pages should be crawled.

```